File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.js
class Solution {
    /**
     * @param {number} prices
     * @return {number}
     */
    maxProfit(prices) {
        let maxProfit = 0

        let left = 0

        for (let right = 0; right < prices.length; right++) {
            if (prices[left] >  prices[right]) {
                left = right
            }
            maxProfit = Math.max(maxProfit, prices[right] - prices[left])
        }

        return maxProfit
    }
}

File: Data Structures & Algorithms/binary-tree-diameter/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        res = 0
        def helper(node):
            nonlocal res
            if not node: return 0

            left = helper(node.left)
            right = helper(node.right)

            res = max(left+right, res)

            return max(left, right) + 1

        helper(root)
        return res
